src/metabase/lib/schema/temporal_bucketing.cljc

```diff
@@ -3,14 +3,70 @@
   (:require
    [metabase.util.malli.registry :as mr]))
 
+;;; units that you can EXTRACT from a date or datetime. These return integers in temporal bucketing expressions.
+(mr/def ::unit.date.extract
```

Comment:
I divided the units up between extraction and truncation units, because they're allowed in different places and because extraction ones cause `:field` clauses to return integers, so it's important to differentiate them